src/js/api/errors.ts

```diff
@@ -161,18 +165,23 @@ export const MSG_TECHNICAL_ERROR = 'technicalError';
  * @param {sourceError} [Error] - The lower level error, caught by the player, which resulted in this error.
  */
 export class PlayerError {
-    constructor(key, code, sourceError = null) {
+    code: ErrorCode;
+    sourceError: Error | PlayerError | null;
```

Comment:
does `sourceError?: Error | PlayerError` work here?

Reply:
That would be equivalent to Error | PlayerError | undefined.